rsciio/usid_hdf5/api.py

```diff
@@ -432,10 +432,6 @@ def file_reader(
             h5_dset, ignore_non_uniform_dims=ignore_non_uniform_dims, lazy=lazy, **kwds
         )
 
-    # At least close the file handle if not lazy load
-    if not lazy:
```

Comment:
When not loading lazily, we still need to close the file.

Reply:
But these two lines are unreachable code:
https://lgtm.com/projects/g/hyperspy/rosettasciio?mode=list&id=py%2Funreachable-statement

The preceding `if/else` are both ending with a `return` statement, so the file closing would not be executed. So I guess, we have to close the file before the return statement.

Reply:
Implemented.

Reply:
ok, done

Reply:
To simplify the code, would it be better to keep only one return at the end of the function?